### PR TITLE
SkipEval exception

### PR DESCRIPTION
**Pull request by sfc-gh-pmardziel**
Items to add to release announcement:
- **SkipEval**: Added exception to be raised by a feedback function to indicate that it should be skipped and not aggregated. This is for use cases were the feedback function is evaluated multiple times on different data and the determination of whether an evaluation on one piece of data should be skipped is done in the feedback implementation itself. Note the related existing feature `if_exists` to skip the entire feedback function for a whole record.

### CODE REVIEW

File: trulens_eval/trulens_eval/feedback/feedback.py [hunk 1]
```diff
@@ -860,13 +889,24 @@ def run(
...
                     result_and_meta, part_cost = mod_base_endpoint.Endpoint.track_all_costs_tally(
                         self.imp, **ins
                     )
 
                     cost += part_cost
+
+                except SkipEval as e:
+                    e.feedback = self
+                    e.ins = ins
+                    skipped_exceptions.append(e)
```
Comment (sfc-gh-chu) [inline trulens_eval/trulens_eval/feedback/feedback.py:906]: any reason we track every skipped exception instance w/ metadata as opposed to (for example) a count?

File: trulens_eval/trulens_eval/feedback/feedback.py [hunk 2]
```diff
@@ -917,6 +957,15 @@ def run(
                     UserWarning,
                     stacklevel=1
                 )
+                # Log any SkipCombination exceptions in case that was the reason
+                # why no results were collected.
+                if len(skipped_exceptions) > 0:
+                    warnings.warn(
+                        "One or more evaluations were skipped because they raised SkipEval (see earlier warnings).",
```
Comment (sfc-gh-chu) [inline trulens_eval/trulens_eval/feedback/feedback.py:964]: can we report the % or count of SkipEvals raised as well?

**Review by sfc-gh-chu — APPROVED** [review]
lgtm w/ some small feedback

File: trulens_eval/trulens_eval/feedback/feedback.py (hunk 1 above)
Comment (sfc-gh-pmardziel) [reply trulens_eval/trulens_eval/feedback/feedback.py:906]: felt simpler to write and more future proof if we want to do more with those exceptions

**Pull request merged**